src/zarr/storage/_cache.py

```diff
@@ -0,0 +1,698 @@
+
+    The cache stores the raw bytes returned by the underlying store, before any
+    decompression or array processing. This means that compressed data remains
+    compressed in the cache, and decompression happens each time the cached data
+    is accessed. This design choice keeps the cache lightweight while still
+    providing significant performance benefits for network-bound operations.
+
+    This store supports both read and write operations. Write operations use a
+    write-through strategy where data is written to both the underlying store
+    and cached locally. The cache automatically invalidates entries when the
+    underlying data is modified to maintain consistency.
+
+    Parameters
+    ----------
+    store : Store
+        The store containing the actual data to be cached.
+    max_size : int
+        The maximum size that the cache may grow to, in number of bytes.
+        This parameter is required to prevent unbounded memory growth.
+
+        Values smaller than your typical chunk size will result in most data
+        being silently excluded from the cache, reducing effectiveness.
+
+    Examples
+    --------
+    The example below wraps a LocalStore with an LRU cache for demonstration::
+
+        >>> import tempfile
+        >>> import zarr
+        >>> from zarr.storage import LocalStore
+        >>>
+        >>> # Create a temporary directory for the example
+        >>> temp_dir = tempfile.mkdtemp()
+        >>> store = LocalStore(temp_dir)
+        >>>
+        >>> # Create some test data first
+        >>> arr = zarr.create((1000, 1000), chunks=(100, 100), store=store, dtype='f4')
+        >>> arr[:] = 42.0
+        >>>
+        >>> # Now wrap with cache for faster access
+        >>> cached_store = zarr.LRUStoreCache(store, max_size=1024 * 1024 * 256)  # 256MB cache
+        >>> cached_arr = zarr.open(cached_store)
+        >>>
+        >>> # First access loads from disk and caches
+        >>> data1 = cached_arr[0:100, 0:100]  # Cache miss
+        >>>
+        >>> # Second access uses cache (much faster for remote stores)
+        >>> data2 = cached_arr[0:100, 0:100]  # Cache hit
+
+    For remote stores where the performance benefit is more apparent::
+
+        >>> from zarr.storage import RemoteStore
+        >>> # remote_store = RemoteStore.from_url("https://example.com/data.zarr")
+        >>> # cached_remote = zarr.LRUStoreCache(remote_store, max_size=2**28)
+
+
+    """
+
+    supports_writes: bool = True
+    supports_deletes: bool = True
+    supports_partial_writes: bool = True
+    supports_listing: bool = True
+
+    root: Path
+
+    def __init__(self, store: Store, *, max_size: int) -> None:
+        if max_size <= 0:
+            raise ValueError("max_size must be a positive integer (bytes)")
+
+        # Always inherit read_only state from the underlying store
+        read_only = getattr(store, "read_only", False)
+
+        # Call parent constructor with read_only parameter
```

Comment:
This comment doesn't describe anything or add any context beyond the actual line of code, so I'd recommend removing.